lib/flatpak-installation.c

```diff
@@ -920,20 +975,104 @@ flatpak_installation_list_installed_refs_for_update (FlatpakInstallation *self,
         g_ptr_array_add (updates, g_object_ref (installed_ref));
     }
 
-  return g_steal_pointer (&updates);
-}
-
 #ifdef FLATPAK_ENABLE_P2P
-static void
-async_result_cb (GObject      *obj,
-                 GAsyncResult *result,
-                 gpointer      user_data)
-{
-  GAsyncResult **result_out = user_data;
-  *result_out = g_object_ref (result);
-}
+  context = g_main_context_new ();
+  g_main_context_push_thread_default (context);
+
+  collection_refv = (OstreeCollectionRef **) g_ptr_array_free (collection_refs, FALSE);
+  collection_refs = NULL;
```

Comment:
A more common approach here in flatpak is
```
 g_ptr_array_free (g_steal_pointer (&collection_refs), FALSE);
```
But this is fine too.

Reply:
Fixed